tests/migrations/test_autodetector.py

```diff
@@ -1229,8 +1229,8 @@ def test_circular_fk_dependency(self):
         # Right number/type of migrations?
         self.assertNumberMigrations(changes, 'testapp', 1)
         self.assertOperationTypes(changes, 'testapp', 0, ["CreateModel", "CreateModel"])
-        self.assertOperationAttributes(changes, "testapp", 0, 0, name="Publisher")
-        self.assertOperationAttributes(changes, "testapp", 0, 1, name="Author")
+        self.assertOperationAttributes(changes, "testapp", 0, 0, name="Author")
+        self.assertOperationAttributes(changes, "testapp", 0, 1, name="Publisher")
```

Comment:
This was the only test that failed with the changes. These are the `ModelState` objects relevant to this test:

```python
    author_with_book = ModelState("testapp", "Author", [
        ("id", models.AutoField(primary_key=True)),
        ("name", models.CharField(max_length=200)),
        ("book", models.ForeignKey("otherapp.Book", models.CASCADE)),
    ])
    publisher_with_book = ModelState("testapp", "Publisher", [
        ("id", models.AutoField(primary_key=True)),
        ("author", models.ForeignKey("otherapp.Book", models.CASCADE)),
        ("name", models.CharField(max_length=100)),
    ])
    book = ModelState("otherapp", "Book", [
        ("id", models.AutoField(primary_key=True)),
        ("author", models.ForeignKey("testapp.Author", models.CASCADE)),
        ("title", models.CharField(max_length=200)),
    ])
```

_(As an aside it looks like the `author` field on `Publisher` should be named `book`.)_

So `Author` and `Publisher` (in `testapp`) both point to `Book` (in `otherapp`) and `Book` points back to `Author`. I don't think that the order of `Author` and `Publisher` matters in this case, but it is probably nicer that `Author` is first.